lib/Core/SQLiteBuildDB.cpp

```diff
@@ -477,19 +469,10 @@ class SQLiteBuildDB : public BuildDB {
     return true;
   }
 
-  static constexpr const char *findIDForKeyInRuleResultsStmtSQL =
-    "SELECT id FROM rule_results "
-    "WHERE key_id == ? LIMIT 1;";
-  sqlite3_stmt* findIDForKeyInRuleResultsStmt = nullptr;
-
   static constexpr const char *insertIntoRuleResultsStmtSQL =
-    "INSERT INTO rule_results VALUES (NULL, ?, ?, ?, ?, ?);";
+    "INSERT OR REPLACE INTO rule_results VALUES (?, ?, ?, ?, ?);";
```

Comment:
Nice! I'm surprised we don't see a win on this PR, but maybe our unit tests don't properly exercise these scenarios.